### ISSUE TITLE
cannot view pdfs

### ISSUE THREAD

**Issue by HemanthJabalpuri** [mpietrzak/apv#167]
Sir when i open the app it shows nothing..
i.e., there are empty directories...
when I open a pdf file from other file managers ..it shows invalid pdf file...

i am running on stock touchwiz 4.4.2 rom

Thanks for lightweight pdf viewer app

**Comment by set-soft** [comment]
I downloaded it from F-Droid https://f-droid.org/es/packages/cx.hell.android.pdfview/
and found the problem is that this tool doesn't ask for the propper permissions (i.e. read the contents of your shared storage)

**Comment by Atralb** [comment]
Yes, as @set-soft explained, the issue is that the process is not in the group owner of common data directories such as `/storage/emulated` where your download folder and such are probably located.

Please @mpietrzak, would you consider fixing this normally small issue ? Your application is precious in that it's the only pdf viewer accessible on FDroid that provides actual inverted color viewing for late night reading. It would be a shame that all your past work wouldn't be useful because of such a tiny bug.

Btw, I would be eager to help you on this fix, however I never developed anything in Java unfortunately so I wouldn't be of great use in this. However, I may be able to help you pinpoint the task needed to correct this.

**Comment by HemanthJabalpuri** [comment]
> I downloaded it from F-Droid https://f-droid.org/es/packages/cx.hell.android.pdfview/
and found the problem is that this tool doesn't ask for the propper permissions (i.e. read the contents of your shared storage)

When I opened this issue, I am using KiKat Android version which we cannot able to manually grant permissions to apps.
I think it is introduced in Marshmallow.

> Your application is precious in that it's the only pdf viewer accessible on FDroid that provides actual negative color viewing for late night reading.

@Atralb
Why don't you use other closed source apps? Privacy issue? Open source thing?
If you want only open-source app then go for https://f-droid.org/packages/org.sufficientlysecure.viewer/ which is also available in PlayStore at https://play.google.com/store/apps/details?id=org.sufficientlysecure.viewer
Or Orion Viewer at https://play.google.com/store/apps/details?id=universe.constellation.orion.viewer
or
https://f-droid.org/en/packages/org.koreader.launcher/
or
https://f-droid.org/en/packages/com.foobnix.pro.pdf.reader/

Closed Source(?) alternatives are at 
https://play.google.com/store/apps/details?id=com.avrapps.pdfviewer

https://play.google.com/store/apps/details?id=bn.tazkir.reader

https://play.google.com/store/apps/details?id=org.ebookdroid
